lib/opentelemetry/src/convert.rs

```diff
@@ -0,0 +1,132 @@
+    fn into_iter(self) -> Self::IntoIter {
+        let resource = self.resource;
+        // convert instrumentation_library_logs(deprecated) into scope_logs
+        let scope_logs: Vec<ScopeLogs> = if !self.scope_logs.is_empty() {
+            self.scope_logs
+        } else {
+            self.instrumentation_library_logs
+                .into_iter()
+                .map(ScopeLogs::from)
+                .collect()
+        };
+
+        scope_logs
+            .into_iter()
+            .map(|scope_log| scope_log.log_records)
+            .flatten()
+            .map(|log_record| {
+                ResourceLog {
+                    resource: resource.clone(),
+                    log_record,
+                }
+                .into()
+            })
+            .collect::<Vec<Event>>()
+            .into_iter()
+    }
+}
+
+impl From<PBValue> for Value {
+    fn from(av: PBValue) -> Self {
+        match av {
+            PBValue::StringValue(v) => Value::Bytes(Bytes::from(v)),
+            PBValue::BoolValue(v) => Value::Boolean(v),
+            PBValue::IntValue(v) => Value::Integer(v),
+            PBValue::DoubleValue(v) => Value::Float(NotNan::new(v).unwrap()),
+            PBValue::BytesValue(v) => Value::Bytes(Bytes::from(v)),
+            PBValue::ArrayValue(arr) => Value::Array(
+                arr.values
+                    .into_iter()
+                    .filter_map(|av| av.value)
+                    .map(|v| v.into())
+                    .collect::<Vec<Value>>(),
+            ),
+            PBValue::KvlistValue(arr) => kv_list_2_value(arr.values),
+        }
+    }
+}
+
+struct ResourceLog {
+    resource: Option<OtelResource>,
+    log_record: LogRecord,
+}
+
+fn kv_list_2_value(arr: Vec<KeyValue>) -> Value {
+    Value::Object(
+        arr.into_iter()
+            .filter_map(|kv| kv.value.map(|av| (kv.key, av)))
+            .fold(BTreeMap::default(), |mut acc, (k, av)| {
+                if let Some(v) = av.value {
+                    acc.insert(k, v.into());
+                }
+                acc
+            }),
+    )
+}
+
+impl From<ResourceLog> for Event {
+    fn from(rl: ResourceLog) -> Self {
+        let mut le = LogEvent::default();
+        // resource
+        if let Some(resource) = rl.resource {
+            le.insert(RESOURCE_KEY, kv_list_2_value(resource.attributes));
+        }
+        le.insert(ATTRIBUTES_KEY, kv_list_2_value(rl.log_record.attributes));
+        if let Some(v) = rl.log_record.body.and_then(|av| av.value) {
+            le.insert(log_schema().message_key(), v);
+        }
+        le.insert(
+            log_schema().timestamp_key(),
+            rl.log_record.time_unix_nano as i64,
+        );
+        le.insert(
+            TRACE_ID_KEY,
+            Value::Bytes(Bytes::from(hex::encode(rl.log_record.trace_id))),
+        );
+        le.insert(
+            SPAN_ID_KEY,
+            Value::Bytes(Bytes::from(hex::encode(rl.log_record.span_id))),
+        );
```

Comment:
I guess those fields will be empty  if there is no tracing involved ? I'm wondering if those very specific field could be ignored if they are empty to prevent any confusion.